### PR TITLE
Clarified the inputs for the PID case study.

### PR DESCRIPTION
**Pull request by theyoucheng**
- specify input parameters for the PID function more clearly
- non-deterministically assign input values at each loop iteration

### CODE REVIEW

**Comment by kroening** [comment]
The test suite seems to only need to feed inputs twice into the controller; why are 6 unwindings needed? What are the inputs used in the remaining 4 iterations?

**Comment by theyoucheng** [comment]
Added the test suite for unwinding the loop six times.

File: doc/html-manual/cover.shtml [hunk 1]
```diff
@@ -70,88 +70,90 @@ is plotted in the Figure below.
...
-49:   /** Estimates the vertical speed */
-50:   estimator_z_dot=climb_pgain * err + desired_climb;
+18:
+19: /** PID function INPUTS */
+20: // The user input: target speed in vertical direction
+21: float desired_climb;
+22: // Vertical speed of the UAV detected by GPS sensor
+23: float estimator_z_dot;
+24:
+25: /** PID funciton OUTPUTS */
```
Comment (kroening) [inline doc/html-manual/cover.shtml]: Please fix typo

File: doc/html-manual/cover.shtml [hunk 2]
```diff
@@ -70,88 +70,90 @@ is plotted in the Figure below.
...
+58:     /** Non-deterministic input values */ 
+59:     desired_climb=nondet_float();
+60:     estimator_z_dot=nondet_float();
+61:
+62:     /** Range of input values */ 
+63:     __CPROVER_assume(desired_climb>=-MAX_CLIMB && desired_climb<=MAX_CLIMB);
+64:     __CPROVER_assume(estimator_z_dot>=-MAX_CLIMB && estimator_z_dot<=MAX_CLIMB);
+65:
+66:     climb_pid_run(); 
+67:
```
Comment (kroening) [inline doc/html-manual/cover.shtml]: Add a __CPROVER_output for the output

File: doc/html-manual/cover.shtml [hunk 3]
```diff
@@ -70,88 +70,90 @@ is plotted in the Figure below.
...
+their values are bounded within a valid range.
 </p>
 
 <p class="justified">
 To demonstrate the automatic test suite generation in CBMC,
 we call the following command
 </p>
 
-<pre><code>cbmc pid.c --cover mcdc --unwind 25 --trace --xml-ui
+<pre><code>cbmc pid.c --cover mcdc --unwind 6 --trace --xml-ui
```
Comment (kroening) [inline doc/html-manual/cover.shtml]: I am not told why "6" is the right number

File: doc/html-manual/cover.shtml [hunk 4]
```diff
@@ -171,20 +173,33 @@ goals at the same time.  Each trace corresponds to a test case.
...
+A test suite consists of a sequence of input parameters that are
+passed to the PID function <code>climb_pid_run</code> at each loop iteration. 
 <pre><code>Test suite:
-T1: desired_climb=-1.000000f, estimator_z_dot=1.000000f
-    (goal id: climb_pid_run.coverage.1)
-T2: desired_climb=1.000000f, estimator_z_dot=-1.000000f
-    (goal id: climb_pid_run.coverage.2)
-T3: desired_climb=0.000000f, estimator_z_dot=0.000000f
-    (goal id: climb_pid_run.coverage.3)
+T1: (desired_climb=-1.000000f, estimator_z_dot=-1.000000f)
```
Comment (kroening) [inline doc/html-manual/cover.shtml]: This test suite does not tell me which numbers to feed in in which iteration

File: doc/html-manual/cover.shtml [hunk 5]
```diff
@@ -70,88 +70,90 @@ is plotted in the Figure below.
...
+56:   while(1)
+57:   {
+58:     /** Non-deterministic input values */ 
+59:     desired_climb=nondet_float();
+60:     estimator_z_dot=nondet_float();
+61:
+62:     /** Range of input values */ 
+63:     __CPROVER_assume(desired_climb>=-MAX_CLIMB && desired_climb<=MAX_CLIMB);
+64:     __CPROVER_assume(estimator_z_dot>=-MAX_CLIMB && estimator_z_dot<=MAX_CLIMB);
+65:
```
Comment (kroening) [inline doc/html-manual/cover.shtml]: There should be two __CPROVER_input commands to record these.

File: doc/html-manual/pid.c [hunk 6]
```diff
@@ -47,25 +48,23 @@ void climb_pid_run()
...
-
 int main()
 {
-  /** Non-deterministic initialisation */ 
-  desired_climb=nondet_float();
-  estimator_z_dot=nondet_float();
-
-  /** Range of initial values of variables */ 
-  __CPROVER_assume(desired_climb>=-MAX_CLIMB && desired_climb<=MAX_CLIMB);
-  __CPROVER_assume(estimator_z_dot>=-MAX_CLIMB && estimator_z_dot<=MAX_CLIMB);
```
Comment (kroening) [inline doc/html-manual/pid.c]: climd_sum_err should be initialized explicitly

**Comment by theyoucheng** [comment]
Thanks Daniel. For now the test suites are manually linked to its coverage goals in this manual. I am going to add an extra script for this later.

**Pull request merged**